### PR TITLE
Add Enterprise Search clients to Elastic docs

### PR DESCRIPTION
**Pull request by chriscressman**
**Do not merge until after the 7.11.0 release**

The Clients team converted their Enterprise Search clients docs from GitHub readmes to Elastic docs.

This PR adds new books for the Enterprise Search Python and Ruby clients, within a new parent page. I followed the pattern established by the Elasticsearch clients.

I also added attributes and aliases for the enterprise search docs, since they were missing (or incorrect).

### CODE REVIEW

File: conf.yaml
Comment (gtback) [inline conf.yaml:50]: I think it would be better if these were all alphabetized (I guess I didn't catch it when `enterprise-search-pubs` was added).

**Review by gtback — APPROVED** [review]
This looks fine to me. 

I'm not a huge fan of the "nested" books, even though it does work fine. With only two (Python and Ruby) I wouldn't be opposed to putting them directly on the Guide home page, but if there are eventually going to be more Enterprise Search clients, it probably makes the most sense to nest them the way you've done it.

**Comment by chriscressman** [comment]
Thanks for the review @gtback :smile: 

I'd like to give the Clients team a preview they can browse. The build log includes the following URL, but it doesn't resolve:

https://docs_2031.docs-preview.app.elstc.co/diff

Do you know what I have to do to get a working preview?

**Comment by gtback** [comment]
@chriscressman Looks like it's working now. Sometimes it takes a minute or two after the build finishes for it to be available on the preview site.

**Comment by chriscressman** [comment]
> but if there are eventually going to be more Enterprise Search clients, it probably makes the most sense to nest them the way you've done it.

Yes, the Clients team will be publishing more Enterprise Search clients in future releases.

File: conf.yaml
Comment (chriscressman) [reply conf.yaml]: :+1: fixed

**Comment by chriscressman** [comment]
@elastic/enterprise-search-clients

This PR adds the docs for the Python and Ruby clients to the centralized docs build.

To preview the changes:

1. Visit https://docs_2031.docs-preview.app.elstc.co/guide/index.html. This is what the docs landing page will look like after this PR is merged.
1. Under _Enterprise Search_, choose **Enterprise Search Clients**. You land on another index page, which includes the current clients. In the future, we can add more clients here.

Under this configuration, the docs build will pick up and publish all changes to the master branch of each client repo.

To _version_ your docs, you'd need to push up version-specific branches in each of your repos (e.g. a _7.11_ repo for this release). If you decide to do that, I'll update this PR.

If you'd rather keep it simple and publish only your master branch, I'll merge this PR as-is.

Let me know which approach you prefer. Either way, after I merge this PR, I'll update the following page in the Enterprise Search docs to point to the new client docs:

https://www.elastic.co/guide/en/enterprise-search/current/programming-language-clients.html

File: conf.yaml
Comment (gtback) [inline conf.yaml:37]: Even though this (apparently) works either way, for consistency I'd add .git extensions
```suggestion
    enterprise-search-python: https://github.com/elastic/enterprise-search-python.git
```

File: conf.yaml
Comment (gtback) [inline conf.yaml:38]:
```suggestion
    enterprise-search-ruby: https://github.com/elastic/enterprise-search-ruby.git
```

**Review by gtback — APPROVED** [review]

**Review by gtback — APPROVED** [review]

**Review by sethmlarson — APPROVED** [review]
@chriscressman Thanks for this, it looks great! @picandocodigo and I will be copying the current documentation to the `7.x` and `7.11` branches so we can have versioned documentation.

**Review by picandocodigo — APPROVED** [review]

**Comment by picandocodigo** [comment]
@chriscressman Both @sethmlarson and I added the docs to `7.x` and `7.11` for versioned documentation :+1:

**Comment by chriscressman** [comment]
@elastic/enterprise-search-clients 

After the current build finishes (assuming it passes), the following URL will preview the docs site with the additional branches for the Enterprise Search clients:

https://docs_2031.docs-preview.app.elstc.co/guide/index.html

Take a look and let me know if everything looks good.

On the live website, the current branch is _7.10_, but the Enterprise Search clients don't have docs for that version. If I would merge this PR now, the master build would fail. Therefore, **I will merge this PR after the 7.11 docs are the current docs** (e.g. on 7.11.0 release day). At that time, this configuration will build successfully when merged to master.

**Comment by sethmlarson** [comment]
@elasticmachine run docs build

**Comment by sethmlarson** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by sethmlarson** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

**Comment by sethmlarson** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by gtback** [comment]
It looks like there are some broken links from Cloud to the Elasticsearch reference; I wonder if that's due to bumping the "current" version to 7.11. 

I'm tempted to let this wait until after the 7.11 release and try again, unless you have any other ideas @chriscressman

**Comment by sethmlarson** [comment]
@gtback I pinged Slack and I believe the issue is resolved now with the broken links being fixed? We'll see with the latest build :)

**Comment by sethmlarson** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by sethmlarson** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by chriscressman** [comment]
@elastic/enterprise-search-clients @gtback 

The release date for 7.11.0 has been moved to Wednesday (or later). I'm trying to arrange coverage for me to be out of office the rest of this week.

Would someone be able to merge this PR for me after the docs release finishes on release day?

The process would be:

1. Wait for the the 7.11.0 docs release to finish (check #mission-control, #techpubs, or just wait until late in the day)
2. Merge master into this PR and wait for a successful build
3. After it's green, merge
4. If you don't see the docs on the website after an hour or so, ask someone in #techpubs to run a full docs rebuild (I'm not sure if this change requires that step or not)

**Comment by gtback** [comment]
The build is failing for the same reason the master branch is failing. Once it's fixed there, I'll merge it in here and it should pass.

**Comment by gtback** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**
